test/api/test_plan_serialization_bwc.cpp

```diff
@@ -0,0 +1,71 @@
+#include "catch.hpp"
+#include "duckdb/common/field_writer.hpp"
+#include "duckdb/common/serializer/buffered_file_reader.hpp"
+#include "duckdb/common/serializer/buffered_file_writer.hpp"
+#include "duckdb/planner/planner.hpp"
+#include "duckdb/parser/parser.hpp"
+#include "test_helpers.hpp"
+#include "tpch-extension.hpp"
+
+using namespace duckdb;
+using namespace std;
+
+string get_file_name(FileSystem &fs) {
+	auto file_name = string("test_plan_serialization_bwc.cpp");
+	auto path = string(__FILE__);
+	return path.replace(path.rfind(file_name), file_name.length(), "serialized_tpch_queries.binary");
+}
+
+TEST_CASE("Generate serialized TPCH file", "[.]") {
+	DuckDB db;
+	Connection con(db);
+	con.Query("CALL dbgen(sf=0.01)");
+	BufferedFileWriter serializer(db.GetFileSystem(), get_file_name(db.GetFileSystem()));
+	serializer.SetVersion(PLAN_SERIALIZATION_VERSION);
+	serializer.Write(serializer.GetVersion());
+	for (int q = 1; q < 22; q++) {
+		auto sql = TPCHExtension::GetQuery(q);
+
+		con.BeginTransaction();
+		Parser p;
+		p.ParseQuery(sql);
+
+		Planner planner(*con.context);
+
+		planner.CreatePlan(move(p.statements[0]));
+		auto plan = move(planner.plan);
+		plan->Serialize(serializer);
+
+		con.Rollback();
+	}
+	serializer.Sync();
+}
+
+TEST_CASE("Test deserialize TPCH from file", "[serialization]") {
```

Comment:
This should be a slow test as well as `dbgen` is quite slow to run, even for low scale factors (as dbgen always generates rather large dictionaries for random value generation regardless of scale factor)

Reply:
good call. Will do.